core/rio/hdt/src/main/java/org/eclipse/rdf4j/rio/hdt/part/HDTPart.java

```diff
@@ -0,0 +1,235 @@
+
+		if (len == BUFLEN) {
+			throw new IOException("Buffer for reading properties exceeded");
+		}
+		return Arrays.copyOf(buf, len + 1);
+	}
+
+	/**
+	 * Get the properties from the input stream, reading at most BUFLEN bytes. The properties are encoded as a
+	 * <code>key=value;</code> string and must be <code>null</code> terminated.
+	 * 
+	 * @param is input stream
+	 * @return key,value map
+	 * @throws IOException
+	 */
+	protected static Map<String, String> getProperties(InputStream is) throws IOException {
+		return mapProperties(readToNull(is));
+	}
+
+	/**
+	 * Get properties as a key, value map
+	 * 
+	 * @param props
+	 * @return
+	 */
+	protected static Map<String, String> mapProperties(byte[] props) {
+		Map<String, String> map = new HashMap<>();
+		if (props == null || props.length == 0) {
+			return map;
+		}
+
+		String strs[] = new String(props, 0, props.length).split(";");
+		for (String str : strs) {
+			String prop[] = str.split("=");
+			if (prop.length == 2) {
+				map.put(prop[0], prop[1]);
+			}
+		}
+		return map;
+	}
+
+	/**
+	 * Get the positive integer value from a property map. Throw an exception when the property is missing, or less than
+	 * 1.
+	 * 
+	 * @param props property map
+	 * @param prop  name of the property
+	 * @param name  display name of the property
+	 * @return positive integer
+	 * @throws IOException
+	 */
+	protected int getIntegerProperty(Map<String, String> props, String prop, String name) throws IOException {
+		int len = 0;
+
+		String str = props.getOrDefault(prop, "0");
+		try {
+			len = Integer.parseInt(str);
+		} catch (NumberFormatException nfe) {
+			throw new IOException(name + " is not an integer: " + str);
+		}
+		if (len < 1) {
+			throw new IOException(name + " is less than 1: " + len);
+		}
+		return len;
+	}
+
+	/**
+	 * Compare the calculated checksum to the expected one.
+	 * 
+	 * @param cis checked input stream
+	 * @param is  (unchecked) input stream
+	 * @param len number of bytes of the checksum
+	 * @throws IOException
+	 */
+	protected static void checkCRC(CheckedInputStream cis, InputStream is, int len) throws IOException {
+		long calc = cis.getChecksum().getValue();
+
+		byte[] checksum = new byte[len];
+		is.read(checksum);
+
+		long expect = 0L;
+		// little-endian to big-endian, e.g. HDT-It stores checksum 7635 as 0x35 0x76 (at least on x86)
+		for (int i = len - 1; i >= 0; i--) {
+			expect <<= 8;
+			expect |= checksum[i] & 0xFF;
+		}
+
+		if (calc != expect) {
+			throw new IOException("CRC does not match: calculated " +
+					Long.toHexString(calc) + " instead of " + Long.toHexString(expect));
+		}
+	}
+
+	/**
+	 * Writes the CRC to the output stream
+	 * 
+	 * @param cos checked output stream
+	 * @param os  output stream
+	 * @param len number of bytes of the checksum
+	 * @throws IOException
+	 */
+	/*
+	 * protected static void writeCRC(CheckedOutputStream cos, OutputStream os, int len) throws IOException { long calc
```

Comment:
Would be best to leave this code out of the file rather than commented until the writer is implemented (if it is used at that point)

Reply:
ok, done